hazelcast/src/test/java/com/hazelcast/map/impl/mapstore/writebehind/WriteBehindEntryStoreQueueReplicationTest.java

```diff
@@ -78,53 +86,59 @@ public void testWriteBehindQueueIsReplicatedWithExpirationTimes() {
     }
 
     @Test
-    public void testWriteBehindQueueIsReplicatedToNewInstancesWithExpirationTimes() {
+    public void queued_entries_with_expirationTimes_are_replicated_when_cluster_scaled_up() {
         final int entryCount = 1000;
-        final int ttlSec = 30;
+        final int ttlSec = 10;
+        final int writeDelaySec = 5;
+        final int backupCount = 1;
         final String mapName = randomMapName();
         TestEntryStore<Integer, Integer> testEntryStore = new TestEntryStore<>();
 
+        Config config = getConfigWithEntryStore(testEntryStore, writeDelaySec, backupCount);
+
         TestHazelcastInstanceFactory factory = createHazelcastInstanceFactory();
-        HazelcastInstance[] instances = new HazelcastInstance[3];
-        instances[0] = factory.newHazelcastInstance(getConfigWithEntryStore(testEntryStore));
-        instances[1] = factory.newHazelcastInstance(getConfigWithEntryStore(testEntryStore));
-        instances[2] = factory.newHazelcastInstance(getConfigWithEntryStore(testEntryStore));
-        IMap<Integer, Integer> map = instances[0].getMap(mapName);
+        HazelcastInstance node1 = factory.newHazelcastInstance(config);
+
+        IMap<Integer, Integer> map = node1.getMap(mapName);
         long expectedExpirationTime = System.currentTimeMillis() + ttlSec * 1000;
         for (int i = 0; i < 1000; i++) {
             map.put(i, i, ttlSec, TimeUnit.SECONDS);
         }
-        factory.terminate(instances[0]);
-        waitAllForSafeState(instances);
-        factory.terminate(instances[1]);
-        waitAllForSafeState(instances);
 
-        instances[0] = factory.newHazelcastInstance(getConfigWithEntryStore(testEntryStore));
-        instances[1] = factory.newHazelcastInstance(getConfigWithEntryStore(testEntryStore));
+        // scale up
+        HazelcastInstance node2 = factory.newHazelcastInstance(config);
+        HazelcastInstance node3 = factory.newHazelcastInstance(config);
 
-        IMap<Integer, Integer> mapFromNewInstance = instances[2].getMap(mapName);
+        IMap<Integer, Integer> mapFromNewInstance = node3.getMap(mapName);
 
-        // assert that entries are still stored in the entry store after original nodes crash
+        // assert that entries are still stored in
+        // the entry store after original nodes crash
         assertTrueEventually(() -> {
             for (int i = 0; i < entryCount; i++) {
                 testEntryStore.assertRecordStored(i, i, expectedExpirationTime, 2000);
-                assertEquals(i, (int) mapFromNewInstance.get(i));
```

Comment:
For my understanding: we remove this assertion here and above because `assertRecordStore` also checks equality? Can't we have the same NPE in that method, if the env is too busy to get here within the expected TTL time? 

If we know upfront that the assertion will fail - and in this case, we know it will fail because too much time elapsed since puts - I think it's better to skip the assertion. WDYT?

Also, since we wait for the safe state, shouldn't the entries already there on `node3` when we get into `assertTrueEventually`? I think we had `assertTrueEventually` here because with `terminate` the cluster needed to notice the dead node, while with graceful shutdown it shouldn't be the case.

Reply:
Removed `assertEquals(i, (int) mapFromNewInstance.get(i));` because it can fail due to the jitter. It is possible that `mapFromNewInstance` can see entry expired even it was persisted to map-store.
  
>we remove this assertion here and above because assertRecordStore also checks equality

Equality check in `assertRecordStore` checks if entry persisted to map-store. It is not same with `mapFromNewInstance.get(i)`.

Reply:
> I think we had assertTrueEventually here because with terminate the cluster needed to notice the dead node, while with graceful shutdown it shouldn't be the case.

I supposed `assertTrueEventually` is needed because of the write behind.

Reply:
Right, my thinking was biased by the removed line :smile:  It all makes sense now, thanks for the clarification :bulb: